tests/unit/test_req.py

```diff
@@ -367,3 +376,132 @@ def test_req_file_no_finder(tmpdir):
+
+
+@pytest.fixture
+def session():
+    return PipSession()
+
+
+@pytest.fixture
+def finder(session):
+    return PackageFinder([], [], session=session)
+
+
+def test_parse_options_from_requirements(finder):
+    pl = parse_line
+
+    assert pl('-i abc') == (OPTION, ('-i', 'abc'))
+    assert pl('-i=abc') == (OPTION, ('-i', 'abc'))
+    assert pl('-i  =  abc') == (OPTION, ('-i', 'abc'))
+
+    assert pl('--index-url abc') == (OPTION, ('--index-url', 'abc'))
+    assert pl('--index-url=abc') == (OPTION, ('--index-url', 'abc'))
+    assert pl('--index-url   =   abc') == (OPTION, ('--index-url', 'abc'))
+
+    with pytest.raises(RequirementsFileParseError):
+        parse_line('--allow-external')
+
+    res = parse_line('--extra-index-url 123')
+    assert res == (OPTION, ('--extra-index-url', '123'))
+
+    next(parse_content('fn', '-i abc', finder=finder), None)
+    assert finder.index_urls == ['abc']
+
+
+def test_parse_flags_from_requirements(finder):
+    assert parse_line('--no-index') == (FLAG, ('--no-index'))
+    assert parse_line('--no-use-wheel') == (FLAG, ('--no-use-wheel'))
+
+    with pytest.raises(RequirementsFileParseError):
+        parse_line('--no-use-wheel true')
+
+    next(parse_content('fn', '--no-index', finder=finder), None)
+    assert finder.index_urls == []
+
+
+def test_get_requirement_options():
+    pro = parse_requirement_options
+
+    res = pro('--aflag --bflag', ['--aflag', '--bflag'])
+    assert res == {'--aflag': '', '--bflag': ''}
+
+    res = pro('--install-options="--abc --zxc"', [], ['--install-options'])
+    assert res == {'--install-options': '--abc --zxc'}
+
+    res = pro('--aflag --global-options="--abc" --install-options="--aflag"',
+              ['--aflag'], ['--install-options', '--global-options'])
+    assert res == {'--aflag': '', '--global-options': '--abc', '--install-options': '--aflag'}
+
+    line = 'INITools==2.0 --global-options="--one --two -3" --install-options="--prefix=/opt"'
+    assert parse_line(line) == (REQUIREMENT, (
+        'INITools==2.0', {
+            '--global-options': '--one --two -3',
+            '--install-options': '--prefix=/opt'
+        }))
+
+
+def test_install_requirements_with_options(tmpdir, finder, session):
+    content = '''
+    INITools == 2.0 --global-options="--one --two -3" \
+                    --install-options="--prefix=/opt"
+    '''
+
+    req_path = tmpdir.join('requirements.txt')
+    with open(req_path, 'w') as fh:
+        fh.write(content)
+
+    req = next(parse_requirements(req_path, finder=finder, session=session))
+
+    req.source_dir = os.curdir
+    with patch.object(subprocess, 'Popen') as popen:
+        try:
+            req.install([])
+        except:
+            pass
+
+        call = popen.call_args_list[0][0][0]
+        for i in '--one', '--two', '-3', '--prefix=/opt':
+            assert i in call
+
+    # TODO: assert that --global-options come before --install-options.
```

Comment:
Could do with a functional test that actually uses the new feature. Probably just create a dummy package that hacks setup.py to report the options passed, and validate that it returns what the requirements file specifies you are passing.